tests/foreman/api/test_setting.py

```diff
@@ -18,228 +18,187 @@
-        prod = entities.Product(organization=org).create()
-        try:
-            for download_policy in download_policies:
-                download_policy_property.value = download_policy
-                download_policy_property.update({'value'})
-                repo = entities.Repository(
-                    product=prod, content_type='yum', organization=org
-                ).create()
-                assert repo.download_policy == download_policy
-                repo.delete()
-        finally:
-            prod.delete()
-            setting_cleanup(setting_name=download_policy_id, setting_value=default_property_value)
+
+
+@run_in_one_thread
+@tier1
+@upgrade
+@pytest.mark.parametrize('setting_update', ['login_text'], indirect=True)
+def test_positive_update_login_page_footer_text(setting_update):
+    """Updates parameter "login_text" in settings
+
+    :id: 91c5373d-b928-419d-8509-761adf5b94b0
+
+    :CaseImportance: Critical
+
+    :expectedresults: Parameter is updated successfully
+
+    :parametrized: yes
+    """
+
+    login_text_value = [login_text for login_text in valid_data_list()][0]
+    setting_update.value = login_text_value
+    setting_update.update({'value'})
+    assert setting_update.value == login_text_value
+
+
+@run_in_one_thread
+@tier2
+@pytest.mark.parametrize('setting_update', ['login_text'], indirect=True)
+def test_positive_update_login_page_footer_text_without_value(setting_update):
+    """Updates parameter "login_text" without any string (empty value)
+
+    :id: 7a56f194-8bde-4dbf-9993-62eb6ab10733
+
+    :expectedresults: login_text has empty value after update
+
+    :parametrized: yes
+    """
+    setting_update.value = ""
+    setting_update.update({'value'})
+    assert setting_update.value == ""
+
+
+@run_in_one_thread
+@tier2
+@pytest.mark.parametrize('setting_update', ['login_text'], indirect=True)
+def test_positive_update_login_page_footer_text_with_long_string(setting_update):
+    """Attempt to update parameter "Login_page_footer_text"
+        with long length string
+
+    :id: fb8b0bf1-b475-435a-926b-861aa18d31f1
+
+    :expectedresults: Parameter is updated
+
+    :parametrized: yes
+    """
+    login_text_value = [login_text for login_text in generate_strings_list(1000)][0]
+    setting_update.value = login_text_value
+    setting_update.update({'value'})
+    assert setting_update.value == login_text_value
+
+
+@pytest.mark.skip_if_open("BZ:1470083")
+@tier2
+@pytest.mark.parametrize('setting_update', ['discovery_hostname'], indirect=True)
+def test_negative_update_hostname_with_empty_fact(setting_update):
+    """Update the Hostname_facts settings without any string(empty values)
+
+    :id: b8e260fc-e263-4292-aa2f-ab37085c7758
+
+    :expectedresults: Error should be raised on setting empty value for
+        hostname_facts setting
+
+    :parametrized: yes
+    """
+    setting_update.value = ""
+    setting_update.update({'value'})
```

Comment:
If that raises exception (and I believe it does), it will bubble up and fail test case. You can use [`pytest.raises`](https://docs.pytest.org/en/stable/reference.html#pytest-raises) to catch it and assert on it.

Reply:
Yes you are right, I have added pytest.raises to catch the error.
```
    with pytest.raises(HTTPError):
        setting_update.update({'value'})
```